src/frontends/tensorflow/src/op/space_to_batch_nd.cpp

```diff
@@ -12,76 +12,66 @@ namespace ov {
 namespace frontend {
 namespace tensorflow {
 namespace op {
-OutputVector translate_batch_nd_and_space_nd_op(const NodeContext& node) {
+
+void normalize_block_shape_pads_crops(const NodeContext& node,
+                                      Output<Node>& block_shape,
+                                      Output<Node>& pads_crops_begin,
+                                      Output<Node>& pads_crops_end) {
     auto input = node.get_input(0);
-    auto block_shape = node.get_input(1);
-    auto crops = node.get_input(2);
+    block_shape = node.get_input(1);
+    auto paddings_crops = node.get_input(2);
 
-    // ng_crops should be of shape N=[ng_input.get_shape()).size()]
-    // But TF's ng_crops input is limited only to the spatial dimensions (neither
-    // batch nor innermost),
-    // which would mean OV inputs have missing ng_crops[0] and ng_crops[N].
-    // Hence, pad ng_crops with zeros at both ends
+    // make sure that paddings_crops and block_shape to have the same type
+    paddings_crops = make_shared<Convert>(paddings_crops, block_shape.get_element_type())->output(0);
 
-    // return with input if rank < 2 as OV's impl doesn't support it
-    const auto& input_pshape = input.get_partial_shape();
-    const auto& block_shape_pshape = block_shape.get_partial_shape();
-    if (input_pshape.rank().is_static() && block_shape_pshape.rank().is_static()) {
-        auto N = input_pshape.rank().get_length();
-        if (N < 2)
-            return {input};
-    } else {
-        // TODO: support dynamic rank
-        TENSORFLOW_OP_VALIDATION(node, false, "Dynamic rank is not supported.");
-    }
+    // compute the input rank and a shape of block_shape [M]
+    // it is needed to normalize block_shape, pads_begin and pads_end
+    auto input_rank = compute_subgraph_scalar_rank(input, element::i32, false);
+    auto M = make_shared<ShapeOf>(block_shape, element::i32);
 
-    auto N = input_pshape.rank().get_length();
+    // compute a number of padding elements required to normalize block_shape
+    // in the beggining and in the end
+    auto num_begin = make_shared<Constant>(element::i32, Shape{1}, 1);
+    auto num_end = make_shared<Subtract>(input_rank, M)->output(0);
```

Comment:
Can rank be lower then M?

Reply:
It is not possible that comes from the TF specification.